save-frontend/src/main/kotlin/com/saveourtool/save/frontend/routing/BasicRouting.kt

```diff
@@ -0,0 +1,242 @@
+/**
+ * All routs for the mobile version of the frontend
+ */
+
+@file:Suppress("FILE_NAME_MATCH_CLASS", "EXTENSION_FUNCTION_WITH_CLASS")
+
+package com.saveourtool.save.frontend.routing
+
+import com.saveourtool.save.domain.TestResultStatus
+import com.saveourtool.save.entities.benchmarks.BenchmarkCategoryEnum
+import com.saveourtool.save.filters.TestExecutionFilters
+import com.saveourtool.save.frontend.components.views.*
+import com.saveourtool.save.frontend.components.views.contests.ContestGlobalRatingView
+import com.saveourtool.save.frontend.components.views.contests.ContestListView
+import com.saveourtool.save.frontend.components.views.contests.UserRatingTab
+import com.saveourtool.save.frontend.components.views.demo.cpgView
+import com.saveourtool.save.frontend.components.views.demo.diktatDemoView
+import com.saveourtool.save.frontend.components.views.projectcollection.CollectionView
+import com.saveourtool.save.frontend.components.views.usersettings.UserSettingsEmailMenuView
+import com.saveourtool.save.frontend.components.views.usersettings.UserSettingsOrganizationsMenuView
+import com.saveourtool.save.frontend.components.views.usersettings.UserSettingsProfileMenuView
+import com.saveourtool.save.frontend.components.views.usersettings.UserSettingsTokenMenuView
+import com.saveourtool.save.frontend.components.views.welcome.WelcomeView
+import com.saveourtool.save.frontend.createRoutersWithPathAndEachListItem
+import com.saveourtool.save.frontend.utils.*
+import com.saveourtool.save.frontend.utils.isSuperAdmin
+import com.saveourtool.save.info.UserInfo
+import com.saveourtool.save.validation.FrontendRoutes.*
+
+import js.core.get
+import org.w3c.dom.url.URLSearchParams
+import react.*
+import react.router.Navigate
+import react.router.Route
+import react.router.Routes
+
+val testExecutionDetailsView = testExecutionDetailsView()
+
+/**
+ * Just put a map: View -> Route URL to this list
+ */
+val basicRouting: FC<AppProps> = FC { props ->
+    val contestView: FC<Props> = withRouter { location, params ->
```

Comment:
I think all these `FC`s need to be declared on top level, not inside another `FC`

Reply:
That’s not possible when you have `props` and need to pass them…